llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -635,14 +639,59 @@ RecurrenceDescriptor::isAnyOfPattern(Loop *Loop, PHINode *OrigPhi,
       return InstDesc(Select, Prev.getRecKind());
   }
 
+  SelectInst *SI = dyn_cast<SelectInst>(I);
```

Comment:
I think this change could do with some comments.

Reply:
Done.